src/analytics/ledger/output_activity.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::HashSet;
+
+use derive_more::{AddAssign, SubAssign};
+
+use super::TransactionAnalytics;
+use crate::types::{
+    ledger::{LedgerOutput, LedgerSpent, MilestoneIndexTimestamp},
+    stardust::block::{
+        output::{AliasId, NftId},
+        Address, Output,
+    },
+};
+
+#[derive(Copy, Clone, Debug, Default, PartialEq, AddAssign, SubAssign)]
+pub struct NftActivityMeasurement {
+    pub created_count: u64,
+    pub transferred_count: u64,
+    pub destroyed_count: u64,
+}
+
+#[derive(Debug)]
+pub struct NftActivityAnalytics {
+    measurement: NftActivityMeasurement,
+}
+
+impl TransactionAnalytics for NftActivityAnalytics {
+    type Measurement = NftActivityMeasurement;
+
+    fn begin_milestone(&mut self, _: MilestoneIndexTimestamp) {
+        self.measurement = NftActivityMeasurement::default();
+    }
+
+    fn handle_transaction(&mut self, inputs: &[LedgerSpent], outputs: &[LedgerOutput]) {
+        let nft_inputs = inputs
+            .iter()
+            .filter_map(|ledger_spent| {
+                if let Output::Nft(nft_output) = &ledger_spent.output.output {
+                    if nft_output.nft_id == NftId::implicit() {
+                        // Convert implicit ids to explicit ids to make all nfts comparable
+                        // Unwrap: acceptable risk
+                        let output_id: iota_types::block::output::OutputId =
+                            ledger_spent.output.output_id.try_into().unwrap();
+                        let nft_id: NftId = iota_types::block::output::NftId::null()
+                            .or_from_output_id(&output_id)
+                            .into();
```

Comment:
This is some functionality that we should provide on our own types to avoid the conversion to `iota_types`.